e2e test suite for prometheus package

JitenPalaparthi opened the pull request:
## What this PR does / why we need it
<!--
Add detailed explanation of what this PR does and why it is
needed.
-->
 Contains  e2e tests for Prometheus addon along with few helper methods. All the below-mentioned steps are automated.

In order to test Prometheus, the below manual steps are to be performed.The same steps are used for automation by using go Ginkgo framework.

- Step-1 ```tanzu package install prometheus.tce.vmware.com```
- Step-2 Make sure that the deployment is ready to serve before going to run tests.
   ```kubectl get deployment deployment.apps/prometheus-server -n prometheus-addon -o jsonpath={..status.conditions[?(@.type=="Available")].status}```
- Step-3 once everything is up and running , the actual tests will do two things.
  - port forward : ```kubectl port-forward deployment.apps/prometheus -n prometheus-addon   56018:9090```
  - check the health of prometheus api: ```crul -I http://127.0.0.1:56018/-/healthy``` .But this step is done in the tests using core golang http pacakge.
  - If the status is 200, that means prometheus is up and running.
- Step-4 : after tests are run , suite will delete prometheus package ```tanzu package delete prometheus.tce.vmware.com```

## Which issue(s) this PR fixes
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes: # [1106](https://github.com/vmware-tanzu/tce/issues/1106)
## Describe testing done for PR
<!--
Example: Created vSphere workload cluster to verify change. 
-->
Tests done on Standalone cluster using ginkgo
- cd to the pacakge path i.e addons/packages/grafana/test/e2e
- ``` ginkgo -v ``` 
## Special notes for your reviewer
<!--
Add any things that reviewers should be aware of as they review
your PR.
Example: Please verify how I handled foo aligns with overall plan.
-->
N/A
## Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below.
-->
N/A
```release-note

```

JitenPalaparthi committed:
add cmdhelper package

JitenPalaparthi committed:
add README.md file

JitenPalaparthi committed:
init go module for prometheus e2e tests

JitenPalaparthi committed:
add util file

util file contains all kubectl and tanzu commands w.r.t prometheus e2e testing

JitenPalaparthi committed:
add prometheus test suite

JitenPalaparthi committed:
add prometheus tests

JitenPalaparthi committed:
fix new line lint issue

jpmcb reviewed (changes requested):
Please make similar changes as requested in: #1068

- The testing module should not be nested in e2e/ but should be one level down in addons/packages/{pacakge-name}/test. The e2e tests can then exist as a go package inside that higher level module.
- Similarly, the readme makes more sense in the same directly as the go mod and not one level up in e2e/

Please reference how this is done in external-dns as this is the model we are approaching testing with: https://github.com/vmware-tanzu/tce/tree/main/addons/packages/external-dns/test

JitenPalaparthi committed:
Merge branch 'vmware-tanzu:main' into prometheus-e2e-tests

Commit:
add prometheus addon package e2e tests

these tests are developed based on ginkgo framework

before the tests are run,cluster should be up and running

tanzu and kubectl cli must be available

Signed-off-by: Jiten Palaparthi <jpalaparthi@vmware.com>

joshrosso commented:
:wave: -- hello, we've decided to close stale PRs, but do acknowledge that this is something we may want to get in eventually. We noticed it currently exists in your fork. If you'd like to re-open a PR with updated E2E tests, we'd love to take a look at them!
